### PR TITLE
Fix leisure-garden-residential area fills

### LINKED ISSUE #6672: Park details disappears at some zoom levels
**Describe the issue**

There appears to be a set of zoom levels where parks/gardens will disappear and then reappear closer:

https://github.com/organicmaps/organicmaps/assets/29158971/325dfc78-7d06-4f50-b647-e70b779215a7

**Steps to reproduce**
1. Go to South Kensington, London
2. Zoom in and out

**System information:**
 - Operating system and its version: Android 14
 - Organic Maps version: 2023.11.17-17-android
 - Device Model: Pixel 6a

### PR DESCRIPTION
**Pull request by pastk**
Fixes #6672

### CODE REVIEW

**Review by vng — COMMENTED** [review]
Keep visible caption at least for z20? It will be indexed in search.

File: data/styles/clear/include/Basemap_label.mapcss [hunk 1]
```diff
@@ -406,7 +406,9 @@ area|z16-[leisure=garden][garden:type!=residential][name]::int_name
 */
 area|z14-[leisure=park][!name],
 area|z14-[landuse=forest][!name],
-area|z16-[leisure=garden][!name],
+area|z16-[leisure=garden][garden:type!=residential][!name],
+{text: none}
+area|z16-[leisure=garden][garden:type=residential],
 {text: none}
```
Comment (vng) [inline data/styles/clear/include/Basemap_label.mapcss:412]: Why so fancy selector? Enough here, no?
```
area|z16-[leisure=garden][garden:type=residential],
{text: none}
```

**Comment by pastk** [comment]
> Keep visible caption at least for z20? It will be indexed in search.

Wow! Yes it would be good to have even residential gardens searchable. But why does it depend on drule's presence?
I think we'd want to display features' names and search for them regardless of presence of a caption drule in the most of cases, right?
E.g. atm we don't have pathtexts for railway lines, but they're searchable and names are displayed in PP.

File: data/styles/clear/include/Basemap_label.mapcss (hunk 1 above)
Comment (pastk) [reply data/styles/clear/include/Basemap_label.mapcss:412]: nope, doesn't work like this...
we need to reset the caption for `leisure-garden`[!name] also

(and overall the "dynamic tags" [name]/[!name]/et. inheritance is crooked and buggy)

**Comment by vng** [comment]
Generator removes name from Features without caption drawing rules by default. Except ones added like exceptions. Do you have examples of railways without captions but searchable?

File: data/styles/clear/include/Basemap_label.mapcss (hunk 1 above)
Comment (vng) [reply data/styles/clear/include/Basemap_label.mapcss:412]: I see no sense here. Remove text style for Features without name :)

AFAIR, we added this [name] selector to draw or not draw symbols. But why captions? Just declare text in base style and it won't be visible anyway for Features without names :)

**Comment by vng** [comment]
Why we don't want to draw text for residential gardens? At least at z20.

AFAIR, we were struggling with symbols and added [name], [!name] selectors. But if it really has a name, why not draw it together with a symbol?

**Comment by pastk** [comment]
> Generator removes name from Features without caption drawing rules by default. Except ones added like exceptions. Do you have examples of railways without captions but searchable?

55.5541254, 47.4961248

File: data/styles/clear/include/Basemap_label.mapcss (hunk 1 above)
Comment (pastk) [reply data/styles/clear/include/Basemap_label.mapcss:412]: 1. Your suggestion leads to removal of z16 & z17 area fills from `leisure-garden` ("dynamic tags" processing is crooked!)
2. Why store and process in runtime the drules that will never be used? [name]/[!name] is used for this case already anyway, why not optimize the captions also?

**Comment by pastk** [comment]
> Why we don't want to draw text for residential gardens? At least at z20.

I don't mind showing them at higher zooms.

But my point is more generic here - we likely have quite a few other cases where a feature might not be searchable because of an absence of a caption drule and its not a wanted or obvious behavior.

File: data/styles/clear/include/Basemap_label.mapcss (hunk 1 above)
Comment (vng) [reply data/styles/clear/include/Basemap_label.mapcss:412]: Well, ok if it is crooked. ATM [!name] && {text: none} looks strange :)

**Review by vng — APPROVED** [review]

**Comment by vng** [comment]
My bad, RemoveNameIfInvisible was commented in the code along time ago.

**Comment by pastk** [comment]
> My bad, RemoveNameIfInvisible was commented in the code along time ago.

So those residential gardens with no caption will be searchable, is that right?

**Pull request merged**